pennylane/decomposition/symbolic_decomposition.py

```diff
@@ -0,0 +1,202 @@
+    # is equivalent to a custom controlled operator should just be the custom controlled
+    # operator given that its base has_adjoint.
+    if (controlled_base_class, num_control_wires) in base_to_custom_ctrl_op():
+        custom_op_type = base_to_custom_ctrl_op()[(controlled_base_class, num_control_wires)]
+        # All gates in base_to_custom_ctrl_op do not have resource params.
+        return {resource_rep(custom_op_type): 1}
+
+    # Handle the general case, here we assume that the adjoint of a controlled gate
+    # whose base has an adjoint that is of its own type, should have the same resource
+    # rep as the controlled gate itself. For example, Adjoint(Controlled(O)) should
+    # have the same resources as Controlled(O) if the adjoint of O is another O.
+    return {resource_rep(base_class, **base_params): 1}
+
+
+@register_resources(_adjoint_controlled_resource)
+def adjoint_controlled_decomp(*_, base, **__):
+    """Decompose the adjoint of a controlled gate whose base has adjoint.
+
+    Precondition:
+    - isinstance(base, qml.ops.Controlled) and base.base.has_adjoint
+
+    """
+    qml.ctrl(
+        base.base.adjoint(),
+        control=base.control_wires,
+        control_values=base.control_values,
+        work_wires=base.work_wires,
+    )
+
+
+@functools.lru_cache()
+def has_adjoint_ops():
+    """A set of operators whose adjoint is an instance of its own type."""
+    return {
+        # identity
+        qml.Identity,
+        qml.GlobalPhase,
+        # non-parametric gates
+        qml.H,
+        qml.X,
+        qml.Y,
+        qml.Z,
+        qml.SWAP,
+        qml.ECR,
+        # single-qubit parametric gates
+        qml.Rot,
+        qml.U1,
+        qml.U2,
+        qml.U3,
+        qml.RX,
+        qml.RY,
+        qml.RZ,
+        qml.PhaseShift,
+        # multi-qubit parametric gates
+        qml.MultiRZ,
+        qml.PauliRot,
+        qml.PCPhase,
+        qml.IsingXX,
+        qml.IsingYY,
+        qml.IsingZZ,
+        qml.IsingXY,
+        qml.PSWAP,
+        qml.CPhaseShift00,
+        qml.CPhaseShift01,
+        qml.CPhaseShift10,
+        # matrix gates
+        qml.QubitUnitary,
+        qml.DiagonalQubitUnitary,
+        qml.BlockEncode,
+        qml.SpecialUnitary,
+        # custom controlled ops
+        qml.CH,
+        qml.CY,
+        qml.CZ,
+        qml.CNOT,
+        qml.CSWAP,
+        qml.CCZ,
+        qml.Toffoli,
+        qml.MultiControlledX,
+        qml.CRX,
+        qml.CRY,
+        qml.CRZ,
+        qml.CRot,
+        qml.ControlledPhaseShift,
+        # arithmetic ops
+        qml.QubitSum,
+        qml.IntegerComparator,
+        # qchem ops
+        qml.SingleExcitation,
+        qml.SingleExcitationMinus,
+        qml.SingleExcitationPlus,
+        qml.DoubleExcitation,
+        qml.DoubleExcitationPlus,
+        qml.DoubleExcitationMinus,
+        qml.OrbitalRotation,
+        qml.FermionicSWAP,
+        # templates
+        qml.CommutingEvolution,
```

Comment:
I guess we can always extend this list, is it tracked somewhere that we want to do make it "complete" :thinking: 
Things that could go in there might be `TrotterProduct` or `QDrift`, maybe? :)

Reply:
There is a more general story for adding full support for all PL operators.